core/include/detray/tools/navigator.hpp

```diff
@@ -297,44 +335,78 @@ class navigator {
     /** Target function of the navigator, finds the next candidates
      *  and set the distance to next
      *
-     * @tparam track_t type of the track (including context)
+     * @tparam stepper_state_t type of stepper state
      *
      * @param navigation is the navigation state
-     * @param track is the current track information
+     * @param stepping the stepper state
      *
      * @return a heartbeat to indicate if the navigation is still alive
      **/
-    template <typename track_t>
+    template <typename stepper_state_t>
     DETRAY_HOST_DEVICE bool target(state &navigation,
-                                   const track_t &track) const {
+                                   stepper_state_t &stepping) const {
+
+        bool heartbeat = true;
+        const auto &track = stepping();
+
+        // get surface candidates when there is no trust
+        if (navigation.trust_level() == e_no_trust) {
+
+            initialize_kernel(navigation, track,
+                              _detector->volumes()[navigation.volume()]);
+
+            stepping.release_step_size();
```

Comment:
Yes, since ```target()``` is always called before ```step()``` in the propagator, we are effectively setting the initial step size estimate to ```1mm``` before every step. Only that this should be the distance to next, with an additional check against ```max_step_size``` of ```1mm```, imo. So ```release_step_size``` should always be called after the navigator updates the distance to next.

Reply:
I changed release_step_size to take the distance to the next surface. would you check it